mathesar/api/viewsets/db_types.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+from db.types.base import db_types_hinted, PostgresType, MathesarCustomType
+from itertools import chain
+
+
+class DbTypeViewSet(viewsets.ViewSet):
+    def list(self, _):
+        known_vanilla_types = (postgres_type for postgres_type in PostgresType)
+        known_custom_types = (mathesar_custom_type for mathesar_custom_type in MathesarCustomType)
+        known_db_types = chain(known_vanilla_types, known_custom_types)
+        data = [
+            {
+                "id": db_type.value,
```

Comment:
This is not how we are representing DB types anywhere else in the API. How we represent types should be consistent with columns in `/api/v0/tables/` (and `/api/v0/databases/<id>/types`, although I assume that API will be deprecated by this one).

Reply:
Hm, this is giving me some pause.

I presume you meant changing `id` to `type` in the below response:

```
GET /api/v0/db_types/
    {
        "id": "_array",
        "hints": null
    },
```

But, looking at the `tables` endpoint, it does use `id` fields for identifying an instansce of some taxonomy. For example, a table is identified by its `id` and so is a column. The column described below says that it has the `type` `INTEGER`, but in the `db_types` response the `id` field signifies a DB type's identifier, not its `type` attribute.
```
GET /api/v0/tables/

          {
            "id": 3,
            "columns": [
                {
                    "id": 29,
                    "type": "INTEGER",
                },
```

Does this make sense?

Reply:
I actually mean the the types are represented in all caps, the `id` nomenclature is fine. We should also make sure usage of spaces and underscores are consistent.

Reply:
Ok, I'll look into this.